src/identity.jl

```diff
@@ -43,25 +78,27 @@ struct IdentityMultiple{T} <: AbstractMatrix{T}
 end
 
 Base.IndexStyle(::Type{<:IdentityMultiple}) = IndexLinear()
-Base.size(𝐼::IdentityMultiple) = 𝐼.n
+Base.size(𝐼::IdentityMultiple) = (𝐼.n, 𝐼.n)
 Base.getindex(𝐼::IdentityMultiple, inds...) = getindex(𝐼.M, inds...)
 Base.setindex!(𝐼::IdentityMultiple, X, inds...) = error("cannot store a value in an `Identity`")
 
-Base.:(*)(x::Number, 𝐼::IdentityMultiple) = IdentityMultiple(x * 𝐼.M, size(𝐼))
+Base.:(*)(x::Number, 𝐼::IdentityMultiple) = IdentityMultiple(x * 𝐼.M, size(𝐼, 1))
 
 function Base.:(+)(𝐼1::IdentityMultiple, 𝐼2::IdentityMultiple)
     @assert size(𝐼1) == size(𝐼2)
-    return IdentityMultiple(𝐼1.M + 𝐼2.M, size(𝐼1))
+    return IdentityMultiple(𝐼1.M + 𝐼2.M, size(𝐼1, 1))
 end
 
 function Base.:(*)(𝐼1::IdentityMultiple, 𝐼2::IdentityMultiple)
     @assert size(𝐼1) == size(𝐼2)
-    return IdentityMultiple(𝐼1.M * 𝐼2.M, size(𝐼1))
+    return IdentityMultiple(𝐼1.M * 𝐼2.M, size(𝐼1, 1))
 end
 
 function Base.show(io::IO, ::MIME"text/plain", 𝐼::IdentityMultiple{T}) where T
-    print(io, "Scalar multiple of the identity matrix of order $(𝐼.n):\n   ", 𝐼.M)
+    print(io, "$(typeof(𝐼)) of value $(𝐼.M.λ) and order $(𝐼.n)")
 end
 
 # callable identity matrix
 LinearAlgebra.I(n::Int, N=Float64) = IdentityMultiple(one(N)*I, n)
+
+LinearAlgebra.I(λ::Number, n::Int, N=Float64) = IdentityMultiple(λ*I, n)
```

Comment:
```suggestion
LinearAlgebra.I(λ::Number, n::Int, N=Float64) = IdentityMultiple(N(λ)*I, n)
```
`N` was not taken into account.